mbf_abstract_nav/src/move_base_action.cpp

```diff
@@ -544,36 +509,57 @@ void MoveBaseAction::actionRecoveryDone(
   }
 }
 
-void MoveBaseAction::actionGetPathReplanningDone(
-    const actionlib::SimpleClientGoalState &state,
-    const mbf_msgs::GetPathResultConstPtr &result)
+void MoveBaseAction::replanningThread()
 {
-  if (!replanning_ || action_state_ != EXE_PATH)
-    return; // replan only while following a path and if replanning is enabled (can be disabled by dynamic reconfigure)
+  ros::Duration update_period(0.001);
+  ros::Time last_replan_time(0.0);
 
-  if (state == actionlib::SimpleClientGoalState::SUCCEEDED)
+  while (ros::ok())
   {
-    ROS_DEBUG_STREAM_NAMED("move_base", "Replanning succeeded; sending a goal to \"exe_path\" with the new plan");
-    exe_path_goal_.path = result->path;
-    mbf_msgs::ExePathGoal goal(exe_path_goal_);
-    action_client_exe_path_.sendGoal(
-        goal,
-        boost::bind(&MoveBaseAction::actionExePathDone, this, _1, _2),
-        boost::bind(&MoveBaseAction::actionExePathActive, this),
-        boost::bind(&MoveBaseAction::actionExePathFeedback, this, _1));
-  }
+    // replan only while following a path and if replanning is enabled (can be disabled by dynamic reconfigure)
+    bool keep_replanning = !replanning_period_.isZero() && action_state_ == EXE_PATH && dist_to_goal_ > 0.1;
 
-  replanning_mtx_.lock();
-  replanning_rate_.sleep();
-  replanning_mtx_.unlock();
+    if (!action_client_get_path_.getState().isDone())
+    {
+      if (action_client_get_path_.waitForResult(update_period))
```

Comment:
why do we wait here only so little? did you had issues with shutting down the thread correctly?

Reply:
no particular reason,,, 1ms sounds reasonable to react fast to changes on state
will inc x5